aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIOProperties.java

```diff
@@ -67,6 +67,14 @@ public class S3FileIOProperties implements Serializable {
    */
   public static final String SSE_TYPE_KMS = "kms";
 
+  /**
+   * S3 DSSE-KMS encryption.
+   *
+   * <p>For more details:
+   * https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingDSSEncryption.html
+   */
+  public static final String DSSE_TYPE_KMS = "kms.dsse";
```

Comment:
The naming `kms.dsse` implies that DSSE will exist as a property namespace under KMS. Are there any other configurations which would actually go under this namespace? If not should we just call it `dsse-kms`?

Reply:
No, there's any other configuration go under this namespace. Changed to `dsse-kms`.